iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tsfile/container/scan/TsFileInsertionScanDataContainer.java

```diff
@@ -373,14 +391,22 @@ private void moveToNextChunkReader() throws IOException, IllegalStateException {
     }
 
     lastMarker = marker;
-    if (Objects.nonNull(timeChunk) && !currentMeasurements.isEmpty()) {
+    if (!recordAlignedChunk(valueChunkList, marker)) {
+      chunkReader = null;
+    }
+  }
+
+  private boolean recordAlignedChunk(final List<Chunk> valueChunkList, final byte marker)
+      throws IOException {
+    if (!valueChunkList.isEmpty()) {
       chunkReader =
           isMultiPage
-              ? new AlignedChunkReader(timeChunk, valueChunkList, filter)
-              : new AlignedSinglePageWholeChunkReader(timeChunk, valueChunkList);
+              ? new AlignedChunkReader(timeChunkList.get(lastIndex), valueChunkList, filter)
+              : new AlignedSinglePageWholeChunkReader(timeChunkList.get(lastIndex), valueChunkList);
       currentIsAligned = true;
-    } else {
-      chunkReader = null;
+      lastMarker = marker;
+      return true;
     }
+    return false;
```

Comment:
The meaning of the return value is not clear enough. Some comments would better be added to the return value of this method. And/Or:
```java
boolean shouldReturn;
// the previous block
return shouldReturn;
```